.Lib9c.Tools/SubCommand/Action.cs

```diff
@@ -22,29 +30,53 @@ public class Action
             )] long blockIndex = 0
         )
         {
-            Type baseType = typeof(Nekoyume.Action.ActionBase);
+            var baseType = typeof(ActionBase);
 
             bool IsTarget(Type type)
             {
-                return baseType.IsAssignableFrom(type) &&
-                    type.GetCustomAttribute<ActionTypeAttribute>() is { } &&
-                    (
-                        !excludeObsolete ||
-                        !(type.GetCustomAttribute<ActionObsoleteAttribute>() is { } aoAttr) ||
-                        aoAttr.ObsoleteIndex > blockIndex
-                    );
+                if (!baseType.IsAssignableFrom(type) ||
+                    type.GetCustomAttribute<ActionTypeAttribute>() is null)
+                {
+                    return false;
+                }
+
+                var isObsoleteTarget = type.GetCustomAttribute<ActionObsoleteAttribute>() is not null;
+                var isObsolete = type.GetCustomAttribute<ActionObsoleteAttribute>() is { } attr &&
+                                 attr.ObsoleteIndex <= blockIndex;
+
+                if (obsoleteOnly) return isObsoleteTarget;
```

Comment:
`isObsoleteTarget` just means that the action has an `ActionObsoleteAttribute` attribute. And it doesn't mean that the action is obsoleted. So it cannot match with the `obsoleteOnly` option's description, `show obsoleted actions only`.